src/service/resource.js

```diff
@@ -396,7 +412,12 @@ export class Resource {
    * @return {!../layout-rect.LayoutRectDef}
    */
   getLayoutBox() {
-    return this.layoutBox_;
+    if (!this.isFixed_) {
+      return this.layoutBox_;
+    }
+    const viewport = this.resources_.getViewport();
+    return moveLayoutRect(this.layoutBox_, viewport.getScrollLeft(),
```

Comment:
Yeah, it should definitely wait for layers work. Or, alternatively we can do a refactoring pass to clarify this API and remove exposing left/top to clients that don't need it.

> I've taken it to mean "give me the absolute position from the origin (0, 0)", which is how everything (until we introduced #isFixed) treated it. This just fixes the value fixed-position's return.

This might be a mis-understanding, but this is not how I read this code. I definitely understand that the current implementation is wrong, but not yet clear that this fixes it.

What does "origin (0, 0)" mean: document origin or viewport origin? From what I see, (assuming we measured the fixed element once at first) if I call this method twice at different scroll positions (e.g. scrollTop=100 and 200) this method will now return different results. I'm not sure if that'd improve things.